crates/matrix-sdk-base/src/response_processors/state_events.rs

```diff
@@ -216,3 +213,30 @@ where
         })
         .unzip()
 }
+
+#[cfg(test)]
+mod tests {
+    use matrix_sdk_test::{async_test, sync_response, SyncResponseFile, DEFAULT_TEST_ROOM_ID};
+    use ruma::user_id;
+
+    use crate::test_utils::logged_in_base_client;
+
+    #[async_test]
+    async fn test_state_events_after_sync() {
+        // Given a room
+        let user_id = user_id!("@u:u.to");
+
+        let client = logged_in_base_client(Some(user_id)).await;
+
+        let response = sync_response(SyncResponseFile::DefaultWithStateInTimeline);
```

Comment:
Ouch, we try to not use the `SyncResponseFile` anymore, and instead use the glorious `EventFactory`, which makes it more ergonomic to create smaller sync responses. See for instance some usage in tests from this module: https://github.com/matrix-org/matrix-rust-sdk/blob/a81ac262d681fb77444ec94e6c4e9fe03bd794bf/crates/matrix-sdk-base/src/read_receipts.rs#L643-L656

And then let's get rid of the new `DefaultWithStateInTimeline`.

Reply:
ack. I've replaced the last commit with a new one in this format.